include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf.h

```diff
@@ -40,6 +40,16 @@ class SpatialOperatorBase;
 template<typename Number>
 class PostProcessorInterface;
 
+template<typename Number>
+class HelpersALE
+{
+public:
+  std::function<void(double const & time)> const move_grid;
```

Comment:
Can you point me to the point where you initialize the ```std::functions```?
I'm asking because I think ```fill_grid_ccordinates(dst, dof_handler)``` and maybe also the other functions could be something which can be defined in a general way, such that we could have a collection of free functions under the namespace ```ALEUtilities``` instead.

Reply:
As written in the introductory statement https://github.com/exadg/exadg/pull/428#issue-1697494278, this has not been implemented so far. The background/reason for the struct/class design is that I only want to pass one object to the time integration class, not a separate object for every lambda function (because we will probably need more than one lambda).

Reply:
Ah, I missed that. 

I agree if we have to use lambdas because the functions are not general enough for free functions. But I currently don't see any reason why free functions are not sufficient. I think we should discuss it when the initialization of the lambdas is implemented :)

Reply:
> But I currently don't see any reason why free functions are not sufficient.

const-correctness !?

The operations to be done on certain objects are not of type `const`. The objects are living in the `Driver`. So if we define the lambda's inside the driver, it is easier to see which parts are able to manipulate objects in a non-const way. In some sense, the lambda functions are "member functions" of `Driver`, manipulating data structures owned by the `Driver`. There is of course some "magic" in the sense that these functions are finally called by the `TimeIntegrator`.

Realizing these functions as true member functions of the `Driver` has the problem that it is not clear who owns the dof vectors to operate on (`Driver` vs. `TimeIntegrator`), as we discussed some time ago ;)

At least this was my idea/motivation behind. Is this understandable?

Reply:
@jh66637 I have now realized the implementation of the lambda functions in the `Driver`. Note that this functionality has (more or less) already been implemented in `Driver::ale_update()` before, and somewhat redundantly in `SpatialOperator` in the previous code version. Zwei Fliegen mit einer Klappe ...

Reply:
After looking at the new commits, I get it :)